src/main/java/com/iridium/iridiumskyblock/managers/IslandManager.java

```diff
@@ -157,7 +159,12 @@ public void teleportHome(@NotNull Player player, @NotNull Island island, int del
      */
     private void teleportHome(@NotNull Player player, @NotNull Island island) {
         player.setFallDistance(0);
-        player.teleport(LocationUtils.getSafeLocation(island.getHome(), island));
+        if (!com.iridium.iridiumskyblock.utils.CheckEnvironment.isPaper()) {
+            player.teleport(LocationUtils.getSafeLocation(island.getHome(), island), PlayerTeleportEvent.TeleportCause.PLUGIN);
```

Comment:
1. Why are you using the full package + class instead of importing it?
2. Just use a variable for the result of `LocationUtils.getSafeLocation(island.getHome(), island)`, no one likes duplicate code